internal/transport/http/handler_test.go

```diff
@@ -0,0 +1,265 @@
+package http
+
+import (
+	"bytes"
+	"github.com/gorilla/mux"
+	"github.com/kiraplenkin/go-musthave-devops/internal/storage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+)
+
+func TestGetAllStats(t *testing.T) {
+	type fields struct {
+		Router  *mux.Router
+		Service *storage.Store
+	}
+	type want struct {
+		code        int
+		contentType string
+		text        string
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		endpoint string
+		want     want
+	}{
+		{
+			name: "OK",
+			fields: fields{
+				Router:  mux.NewRouter(),
+				Service: storage.NewStorage(),
+			},
+			endpoint: "/",
+			want: want{
+				code:        http.StatusOK,
+				contentType: "text/plain; charset=utf-8",
+				text:        "{map[]}",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler := &Handler{
+				Router:  tt.fields.Router,
+				Service: tt.fields.Service,
+			}
+			req := httptest.NewRequest(http.MethodGet, tt.endpoint, nil)
+			w := httptest.NewRecorder()
+			h := http.HandlerFunc(handler.GetAllStats)
+			h.ServeHTTP(w, req)
+			res := w.Result()
+			assert.Equal(t, tt.want.code, res.StatusCode)
+			assert.Equal(t, tt.want.contentType, res.Header.Get("Content-Type"))
+			b, err := ioutil.ReadAll(res.Body)
+			require.NoError(t, err)
+			assert.Equal(t, tt.want.text, string(b))
+			err = res.Body.Close()
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestGetStats(t *testing.T) {
+	type fields struct {
+		Router  *mux.Router
+		Service *storage.Store
+	}
+	type want struct {
+		code        int
+		contentType string
+		text        string
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		endpoint string
+		want     want
+	}{
+		//{
```

Comment:
Тут вставлю, что писал по поводу теста.

Посмотри еще вот в эту сторону

```go
// endpoints_test.go
func TestMetricsHandler(t *testing.T) {
    tt := []struct{
        routeVariable string
        shouldPass bool
    }{
        {"goroutines", true},
        {"heap", true},
        {"counters", true},
        {"queries", true},
        {"adhadaeqm3k", false},
    }

    for _, tc := range tt {
        path := fmt.Sprintf("/metrics/%s", tc.routeVariable)
        req, err := http.NewRequest("GET", path, nil)
        if err != nil {
            t.Fatal(err)
        }

        rr := httptest.NewRecorder()
	
	// Need to create a router that we can pass the request through so that the vars will be added to the context
	router := mux.NewRouter()
        router.HandleFunc("/metrics/{type}", MetricsHandler)
        router.ServeHTTP(rr, req)

        // In this case, our MetricsHandler returns a non-200 response
        // for a route variable it doesn't know about.
        if rr.Code == http.StatusOK && !tc.shouldPass {
            t.Errorf("handler should have failed on routeVariable %s: got %v want %v",
                tc.routeVariable, rr.Code, http.StatusOK)
        }
    }
}
```